barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/goblin_translator_circuit_builder.hpp

```diff
@@ -8,13 +8,71 @@
  * @copyright Copyright (c) 2023
  *
  */
+#include "barretenberg/common/constexpr_utils.hpp"
 #include "barretenberg/ecc/curves/bn254/fq.hpp"
+#include "barretenberg/numeric/uint256/uint256.hpp"
 #include "barretenberg/proof_system/arithmetization/arithmetization.hpp"
+#include "barretenberg/proof_system/op_queue/ecc_op_queue.hpp"
 #include "circuit_builder_base.hpp"
 #include <array>
 #include <cstddef>
+#include <cstdlib>
+#include <iterator>
+#include <tuple>
 namespace proof_system {
-class GoblinTranslatorCircuitBuilder : CircuitBuilderBase<arithmetization::GoblinTranslator> {
+/**
+ * @brief GoblinTranslatorCircuitBuilder creates a circuit that evaluates the correctness of the evaluation of
+ * EccOpQueue in Fq while operating in the Fr scalar field
+ *
+ * @details Goblin Translator Circuit Builder builds a circuit the purpose of which is to calculate the batched
+ * evaluation of 5 polynomials in non-native field represented through coefficients in 4 native polynomials (op,
+ * x_lo_y_hi, x_hi_z_1, y_lo_z_2):
+ *
+ * OP | X_LO | X_HI | Y_LO
+ * 0  | Y_HI | Z1   | Z2
+ *
+ *  OP is supposed to be { 0, 1, 2, 3, 4, 8 }. X_LO and Y_LO need to be < 2¹³⁶, X_HI and Y_LO < 2¹¹⁸, Z1 and Z2 < 2¹²⁸.
```

Comment:
Where do these op values come from? Do these not correspond to the 4 op codes defined in EccOpCode?

Reply:
This is from a talk with Zac. I need to add this to the EccOpQueue.